src/udev/udev-builtin-input_id.c

```diff
@@ -211,6 +211,8 @@ static bool test_pointers(struct udev_device *dev,
                 has_joystick_axes_or_buttons = test_bit(button, bitmask_key);
         for (button = BTN_TRIGGER_HAPPY1; button <= BTN_TRIGGER_HAPPY40 && !has_joystick_axes_or_buttons; button++)
                 has_joystick_axes_or_buttons = test_bit(button, bitmask_key);
+        for (button = BTN_DPAD_UP; button <= BTN_DPAD_RIGHT && !has_joystick_axes_or_buttons; button++)
+                has_joystick_axes_or_buttons = test_bit(button, bitmask_key);
```

Comment:
#ifdef it?

How ancient input headers do you need to be missing the button defines?